### PR TITLE
AEA-542 Logger.error replaced with ClickException for correct handling.

### PR DESCRIPTION
**Pull request by panasevychol**
## Fixes

Logger.error replaced with ClickException for correct handling.

## Types of changes

What types of changes does your code introduce to agents-aea?
_Put an `x` in the boxes that apply_

- [x] Bugfix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)

## Checklist

_Put an `x` in the boxes that apply._

- [ ] I have read the [CONTRIBUTING](../CONTRIBUTING.md) doc
- [ ] I am making a pull request against the `develop` branch (left side). Also you should start your branch off our `develop`.
- [ ] Lint and unit tests pass locally with my changes
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] I have checked that code coverage does not decrease.
- [ ] I have checked that the documentation about the `aea cli` tool works
- [ ] I have added necessary documentation (if appropriate)
- [ ] Any dependent changes have been merged and published in downstream modules

### CODE REVIEW

File: aea/cli/create.py [hunk 1]
```diff
@@ -160,13 +157,10 @@ def create(click_context, agent_name, author, local):
...
-        logger.error("Directory already exist. Aborting...")
-        sys.exit(1)
+        raise click.ClickException("Directory already exist. Aborting...")
     except ValidationError as e:
-        logger.error(str(e))
+        raise click.ClickException(str(e))
         shutil.rmtree(agent_name, ignore_errors=True)
-        sys.exit(1)
     except Exception as e:
         logger.exception(e)
```
Comment (DavidMinarsch) [inline aea/cli/create.py:165]: Missing!

File: aea/cli/create.py [hunk 2]
```diff
@@ -160,13 +157,10 @@ def create(click_context, agent_name, author, local):
         _add_item(click_context, "skill", DEFAULT_SKILL)
 
     except OSError:
-        logger.error("Directory already exist. Aborting...")
-        sys.exit(1)
+        raise click.ClickException("Directory already exist. Aborting...")
     except ValidationError as e:
-        logger.error(str(e))
+        raise click.ClickException(str(e))
```
Comment (DavidMinarsch) [inline aea/cli/create.py:162]: wrong order

File: aea/cli/generate.py [hunk 3]
```diff
@@ -131,30 +126,27 @@ def _generate_item(click_context, item_type, specification_path):
...
-        logger.error(
+        raise click.ClickException(
             "A {} with this name already exists. Please choose a different name and try again.".format(
                 item_type
             )
         )
-        sys.exit(1)
     except ProtocolSpecificationParseError as e:
-        logger.error(
+        raise click.ClickException(
```
Comment (DavidMinarsch) [inline aea/cli/generate.py:135]: wrong order

File: aea/cli/generate.py [hunk 4]
```diff
@@ -131,30 +126,27 @@ def _generate_item(click_context, item_type, specification_path):
...
     except Exception as e:
         logger.debug("Exception thrown: " + str(e))
-        logger.error(
-            "There was an error while generating the protocol. The protocol is NOT generated."
-        )
         shutil.rmtree(
             os.path.join(item_type_plural, protocol_spec.name), ignore_errors=True
         )
-        sys.exit(1)
+        raise click.ClickException(
```
Comment (DavidMinarsch) [inline aea/cli/generate.py:147]: Remove debug and print exception message here

File: aea/cli/common.py [hunk 5]
```diff
@@ -383,8 +378,10 @@ def convert(self, value, param, ctx):
             # everything ok - return the parameter to the command
             return value
         except Exception:
-            logger.error("The name provided is not a path to an AEA project.")
-            self.fail(value, param, ctx)
+            # self.fail(value, param, ctx)
```
Comment (DavidMinarsch) [inline aea/cli/common.py:381]: do we still need self.fail? if so, remove commenting, else remove comment altogether.

File: aea/cli/generate.py (hunk 4 above)
Comment (DavidMinarsch) [reply aea/cli/generate.py:147]: Sorry I think you misunderstood, the exception message should be moved inside the click exception, i.e.: `"There was an error while generating the protocol. The protocol is NOT generated. Exception thrown: " + str(e)`

Comment (panasevychol) [reply aea/cli/generate.py:147]: Oh yes, now I see. Will do.

File: aea/cli/common.py [hunk 6]
```diff
@@ -191,20 +190,18 @@ def try_to_load_agent_config(ctx: Context, is_exit_on_except: bool = True) -> No
             logging.config.dictConfig(ctx.agent_config.logging_config)
     except FileNotFoundError:
         if is_exit_on_except:
-            logger.error(
+            raise click.ClickException(
                 "Agent configuration file '{}' not found in the current directory.".format(
                     DEFAULT_AEA_CONFIG_FILE
                 )
             )
```
Comment (marcofavorito) [inline aea/cli/common.py:197]: I think this is a good use case for:
```
raise ExceptionX(...) from e
```

see https://docs.python.org/3/reference/simple_stmts.html#the-raise-statement

Comment (marcofavorito) [reply aea/cli/common.py:197]: why don't use `AEAException` which can then be re-raised later as a `ClickException`? So we can  distinguish between exception for our logic and exceptions outside it

File: aea/cli/generate.py [hunk 7]
```diff
@@ -131,30 +126,27 @@ def _generate_item(click_context, item_type, specification_path):
...
-        logger.error(
+        shutil.rmtree(
+            os.path.join(item_type_plural, protocol_spec.name), ignore_errors=True
+        )
+        raise click.ClickException(
             "The following error happened while parsing the protocol specification: "
             + str(e)
         )
+    except Exception as e:
+        print("Exception thrown: " + str(e))
```
Comment (DavidMinarsch) [inline aea/cli/generate.py:143]:
```suggestion
```

File: aea/cli/generate.py [hunk 8]
```diff
@@ -131,30 +126,27 @@ def _generate_item(click_context, item_type, specification_path):
...
         shutil.rmtree(
             os.path.join(item_type_plural, protocol_spec.name), ignore_errors=True
         )
-        sys.exit(1)
-    except Exception as e:
-        logger.debug("Exception thrown: " + str(e))
-        logger.error(
+        raise click.ClickException(
             "There was an error while generating the protocol. The protocol is NOT generated."
```
Comment (DavidMinarsch) [inline aea/cli/generate.py:148]:
```suggestion
            "There was an error while generating the protocol. The protocol is NOT generated. Exception: " + str(e)
```

File: aea/cli/common.py (hunk 5 above)
Comment (DavidMinarsch) [inline aea/cli/common.py:381]:
```suggestion
```

**Pull request merged**